### PR TITLE
2291223 entity browser

### PR DESCRIPTION
**Pull request by slashrsm**
D.O. issue https://www.drupal.org/node/2291223

### CODE REVIEW

File: src/Annotation/EntityBrowserTab.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\entity_browser\Annotation\EntityBrowserDisplay.
```
Comment (davereid) [inline src/Annotation/EntityBrowserTab.php]: EntityBrowserTab

**Comment by Berdir** [comment]
Basics look fine to me. Config schema is missing for the tabs (one more things you should be able to steal from page_manager conceptually).

You might want to consider to also use a plugin bag for the display and the other plugin types, as you get certain things like the dependency handling for free then.

One other thing that might be easier if you include it already is that each plugin might be configurable, so you should be able to store configuration for each. You could have display and display_configuration or just display, which would have the ID together with the settings. There's a DefaultSinglePluginBag that would be useful for this I guess.

**Comment by davereid** [comment]
Yes we should definitely be using PluginBags to collect the actual widget plugins, order, and configuration of each tab plugin.

**Comment by slashrsm** [comment]
OK. Added configuration schema and plugin bags for all plugin types.

**Comment by slashrsm** [comment]
Will merge this as other PRs depend on this. Feel free to keep commenting and open follow-up for any fixes that are needed.

**Pull request merged**